tests/rules/taint_decl_in_if_cond1.cpp

```diff
@@ -0,0 +1,34 @@
+void test1() {
+    void *data = get_data();
+
+    source(data);
+    // assume that any call with `&data` will
+    // sanitize `data`
+    maybe_realloc_data(&data);
+    // ok: source-sink-while
+    sink(data);
+}
+
+void test2() {
+    void *data = get_data();
+
+    // BUG: Declaration in `while` loop does not sanitize
```

Comment:
This comment makes me think that it's a TODO that will be fixed later. Is the comment here to indicate that this used to be a bug but it's fixed now?

Reply:
I just copy-pasted the example as it was reported, but I get your point, I'm fixing it.